src/util/multilines-scaler.ts

```diff
@@ -0,0 +1,93 @@
+import { Range, Ranges, IMultiDataPoint, INumericDataPoint } from '../model/types'
+import { ComputedChartAxis } from '../model/computed-chart-axis'
+import { xAxisTimeFormats } from './utility'
+import { ChartLayout } from '../model/chart-layout'
+
+declare let d3: any;
+
+type MaybeRange = Range | undefined;
```

Comment:
I like this new typescript feature; Basically giving us an Option monad in a simpler/lighter syntax (and only compile-time)